ansible_base/jwt_consumer/common/auth.py

```diff
@@ -153,14 +157,22 @@ def map_user_fields(self):
             old_value = getattr(self.user, attribute, None)
             new_value = self.token.get('user_data', {}).get(attribute, None)
             if old_value != new_value:
+                if attribute == "username" and get_user_model().objects.filter(username=new_value).exists():
+                    logger.warning(
+                        f"Renaming user {old_value} to {new_value} would result in a duplicate key error. "
+                        "Please make sure the sync task is running to prevent this warning in the future."
+                    )
+                    continue
                 if attribute == 'is_superuser' and new_value is False:
                     continue
                 logger.debug(f"Changing {attribute} for {self.user.username} from {old_value} to {new_value}")
                 setattr(self.user, attribute, new_value)
                 user_needs_save = True
         if user_needs_save:
-            logger.info(f"Saving user {self.user.username}")
-            self.user.save()
+            with no_reverse_sync():
+                logger.info(f"Saving user {self.user.username}")
+
+                self.user.save()
```

Comment:
We could do the same update_fields here as well by making user_needs_saves an array instead of a bool.

Reply:
I'd rather not mess with that. This is how it was before and we know that worked.